src/librbd/internal.cc

```diff
@@ -2838,6 +2850,51 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force,
       return -EINVAL;
     }
 
+    // is mirroring  enabled for the child?
```

Comment:
Should set the current state to disabling before starting to check for children.